plugins/flytekit-pydantic/flytekitplugins/pydantic/object_store.py

```diff
@@ -0,0 +1,107 @@
+import uuid
+from typing import Any, Dict, Type, TypeVar, cast
+
+import pandas as pd
+import torch.nn as nn
+from typing_extensions import Annotated, NewType
+
+from flytekit.core import context_manager, type_engine
+from flytekit.models import literals
+from flytekit.types import directory
+from flytekit.types.file import file
+
+PYDANTIC_SUPPORTED_FLYTE_TYPES = (
+    nn.Module,
+    pd.DataFrame,
+    file.FlyteFile,
+    directory.FlyteDirectory,
+    # TODO - add all supported types
+)
+
+LiteralObjID = Annotated[str, "Key for unique object in literal map."]
+PythonType = TypeVar("PythonType")  # target type of the deserialization
+
+
+class FlyteObjectStore:
+    """
+    This class is an intermediate store for python objects that are being serialized/deserialized.
+
+    On serialization of a basemodel, flyte objects are serialized and stored in this object store.
+    On deserialization of a basemodel, flyte objects are deserialized from this object store.
+    """
+
+    _literal_store: Dict[LiteralObjID, literals.Literal] = {}
+
+    def __setattr__(self, name: str, value: Any) -> None:
+        raise Exception("Attributes should not be set on the FlyteObjectStore.")
```

Comment:
Question: Is this required to avoid a problem? Typically one wouldn't implement this for a class in python.

Reply:
Not necessary. 

The goal was to emphasize that this class's function is essentially to serve as a namespace with only a few auxiliary methods for storing/retrieving objects. Given this is somewhat less generic than how most classes are used, I considered it beneficial to insert this sort of defensive mechanism, analogous to the init block I implemented.

It's a good sign that you're questioning this. It suggests that the class's purpose is indeed coming across clearly. Let me know whichever way you prefer it and ill make it so